File: test/internal/agentapi/session.go
package agentapi

import (
	"encoding/json"
	"os"
	"path/filepath"
	"strings"
)

type sessionState struct {
	TokenName string `json:"token_name"`
}

func sessionDir(configDir string) string {
	return filepath.Join(configDir, ".agent-api")
}

func sessionFile(configDir string) string {
	return filepath.Join(sessionDir(configDir), "session.json")
}

func GetSessionTokenName(configDir string) string {
	path := sessionFile(configDir)
	data, err := os.ReadFile(path)
	if err != nil {
		return ""
	}

	var state sessionState
	if err := json.Unmarshal(data, &state); err != nil {
		return ""
	}

	return strings.TrimSpace(state.TokenName)
}

func SetSessionTokenName(configDir string, tokenName string) (string, error) {
	if strings.TrimSpace(tokenName) == "" {
		return "", NewAppError("ERR_TOKEN_NOT_FOUND", "Token name cannot be empty.")
	}

	dir := sessionDir(configDir)
	if err := os.MkdirAll(dir, 0o755); err != nil {
		return "", NewAppError("ERR_CONTEXT_INVALID", "Failed to create session directory.")
	}

	path := sessionFile(configDir)
	payload, err := json.MarshalIndent(sessionState{TokenName: tokenName}, "", "  ")
	if err != nil {
		return "", NewAppError("ERR_CONTEXT_INVALID", "Failed to serialize session state.")
	}

	if err := os.WriteFile(path, payload, 0o644); err != nil {
		return "", NewAppError("ERR_CONTEXT_INVALID", "Failed to write session state.")
	}

	return path, nil
}

File: test/agent_api/session.py
from __future__ import annotations

from pathlib import Path
import json

from .errors import AgentApiError


def _session_dir(config_dir: Path) -> Path:
    return config_dir / ".agent-api"


def _session_file(config_dir: Path) -> Path:
    return _session_dir(config_dir) / "session.json"


def get_session_token_name(config_dir: Path) -> str | None:
    path = _session_file(config_dir)
    if not path.exists():
        return None

    try:
        content = json.loads(path.read_text(encoding="utf-8"))
    except json.JSONDecodeError:
        return None

    token_name = content.get("token_name")
    if isinstance(token_name, str) and token_name.strip():
        return token_name
    return None


def set_session_token_name(config_dir: Path, token_name: str) -> Path:
    if not token_name.strip():
        raise AgentApiError("ERR_TOKEN_NOT_FOUND", "Token name cannot be empty.")

    directory = _session_dir(config_dir)
    directory.mkdir(parents=True, exist_ok=True)
    path = _session_file(config_dir)
    payload = {"token_name": token_name}
    path.write_text(json.dumps(payload, indent=2), encoding="utf-8")
    return path
